firefox-ios/Client/Frontend/Browser/Tabs/Views/Animation/TabAnimation.swift

```diff
@@ -152,19 +152,45 @@ extension TabTrayViewController: BasicAnimationControllerDelegate {
         finalFrame: CGRect,
         selectedTab: Tab
     ) {
-        // BVC snapshot animates to the cell
+        // Snapshot of the BVC view
         let bvcSnapshot = UIImageView(image: browserVC.view.snapshot)
         bvcSnapshot.layer.cornerCurve = .continuous
+        bvcSnapshot.layer.cornerRadius = ExperimentTabCell.UX.cornerRadius
         bvcSnapshot.clipsToBounds = true
+        bvcSnapshot.contentMode = .scaleAspectFill
 
-        // This background view is needed for animation between the tab tray and the bvc
+        // Wrap bvcSnapshot in a container to support external border
+        let snapshotContainer = UIView(frame: bvcSnapshot.frame)
+        snapshotContainer.layer.cornerRadius = bvcSnapshot.layer.cornerRadius
+        snapshotContainer.layer.cornerCurve = .continuous
+        snapshotContainer.clipsToBounds = false
+        bvcSnapshot.frame = snapshotContainer.bounds
+
+        // Create border layer
+        let theme = retrieveTheme()
+        let borderColor = selectedTab.isPrivate ? theme.colors.borderAccentPrivate : theme.colors.borderAccent
+        let borderLayer = CAShapeLayer()
+        borderLayer.path = UIBezierPath(
+            roundedRect: snapshotContainer.bounds,
+            cornerRadius: ExperimentTabCell.UX.cornerRadius
+        ).cgPath
+        borderLayer.strokeColor = borderColor.cgColor
+        borderLayer.fillColor = UIColor.clear.cgColor
+        borderLayer.lineWidth = 0
+        borderLayer.opacity = 0
```

Comment:
Wondering if we should add a `UX` struct for those values. Some other values in this `TabAnimation` class could benefit as well I think